sql/hive/src/test/scala/org/apache/spark/sql/hive/LogicalPlanToSQLSuite.scala

```diff
@@ -105,6 +105,27 @@ class LogicalPlanToSQLSuite extends SQLBuilderTest with SQLTestUtils {
     checkHiveQl("SELECT id FROM t0 UNION ALL SELECT CAST(id AS INT) AS id FROM t0")
   }
 
+  test("union distinct") {
+    checkHiveQl("SELECT * FROM t0 UNION SELECT * FROM t0")
+  }
+
+  // Parser is unable to parse the following query:
+  // SELECT  `u_1`.`id`
+  // FROM (((SELECT  `t0`.`id` FROM `default`.`t0`)
+  // UNION ALL (SELECT  `t0`.`id` FROM `default`.`t0`))
+  // UNION ALL (SELECT  `t0`.`id` FROM `default`.`t0`)) AS u_1
+  ignore("three-child union") {
```

Comment:
@gatorsmile Yeah we should fix this. The parser does not like the nesting within the UNION ALL. It parses fine if you remove the first nested `UNION ALL`:

    SELECT  `u_1`.`id`
    FROM ((SELECT  `t0`.`id` FROM `default`.`t0`)
    UNION ALL (SELECT  `t0`.`id` FROM `default`.`t0`)
    UNION ALL (SELECT  `t0`.`id` FROM `default`.`t0`)) AS u_1